bundles/ch.elexis.molemax/src/ch/elexis/molemax/data/Tracker.java

```diff
@@ -556,11 +555,15 @@ private static String makeFilename(final int x, final int y, final int w, final
 	}
 
 	public static String makeDescriptorImage(final Patient p) {
-		String input = CoreHub.localCfg.get(MolemaxImagePrefs.CUSTOM_BASEDIR, StringUtils.EMPTY);
-
+		String basePath = CoreHub.localCfg.get(MolemaxPrefs.BASEDIR, StringUtils.EMPTY);
+		String customPath = CoreHub.localCfg.get(MolemaxImagePrefs.CUSTOM_BASEDIR, StringUtils.EMPTY);
+		if (StringUtils.isBlank(basePath) || StringUtils.isBlank(customPath)) {
+			log.log("Fehler: Basispfad oder benutzerdefinierter Pfad ist nicht gesetzt.", Log.WARNINGS);
+			return StringUtils.EMPTY;
+		}
 		StringBuilder ret = new StringBuilder();
-		ret.append(CoreHub.localCfg.get(MolemaxPrefs.BASEDIR, StringUtils.EMPTY)).append(File.separator);
-		String[] pathSegments = input.split("/");
+		ret.append(basePath).append(File.separator);
+		String[] pathSegments = customPath.split("/");
```

Comment:
Hier müsste vmtl. auch File.separator verwendet werden.

Wobei bitte überprüfen ob das dann klappt. Mir ist nicht ganz klar weshalb das mit / auf Windows funktioniert wo eigentlich \ der separator ist.

Reply:
Thomas du hast recht, dass an dieser Stelle File.separator verwendet werden sollte. Allerdings ist zu beachten, dass auf Windows-Systemen der Backslash (\) als Trennzeichen dient, welcher in regulären Ausdrücken ein Escape-Zeichen ist. Daher müsste man in regulären Ausdrücken doppelte Backslashes (\\\\) verwenden. Um diese Plattformabhängigkeit zu umgehen und den Code universell einsetzbar zu machen, habe ich eine Methode implementiert, die Pattern.quote(File.separator) verwendet. Diese Methode stellt sicher, dass das richtige Trennzeichen in einem Format-sicheren String für reguläre Ausdrücke auf allen Betriebssystemen korrekt verwendet wird.